### ISSUE TITLE
QnAMaker.ConvertLegacyResult() does not properly handle v2.0 QnA API, only v3.0

### ISSUE THREAD

**Issue by Zerryth** [Microsoft/botbuilder-dotnet#1285]
## Version
dotnet 4.2.0

## Describe the bug
Right now `QnAMaker` class has `ConvertLegacyResult()`, which should take the response from a call to a legacy qna endpoint, and convert it to a QueryResult that matches the shape of response that come from qna v4.0.

` var isLegacyProtocol = _endpoint.Host.EndsWith("v2.0") || _endpoint.Host.EndsWith("v3.0");`

`ConvertLegacyResult()` only works if the endpoint ends in `v3.0`--when it ends in `v2.0`, it will break, because responses that come from `v2.0` don't have the following properties:
* `QnaId`
* `Metadata`
* `Source`

and the code tries to create a `QueryResult`:

```
new QueryResult
                    {
                        // The old version of the protocol returns the "id" in a field called "qnaId"
                        Id = answer.QnaId,
                        Answer = answer.Answer,
                        Metadata = answer.Metadata,
                        Score = answer.Score,
                        Source = answer.Source,
                        Questions = answer.Questions,
                    }
```

will throw an error trying to access members that don't exist

## Expected behavior
`QnAMaker.ConvertLegacyResults()` should convert the results of a `v2.0` QnA response if we are including it as a pathway in our `_isLegacyProtocol` check

Should discuss what values you would want the `QnAMaker` class to actually default to if `v2.0`
for example, do we want the values to be empty `""`?  
* might work for `QueryResult.Source`, but probably not very viable for `QueryResult.Id`

[bug]

**Comment by cleemullins** [comment]
IS this a serious enough issues that we should fix and issue a 4.2.1 patch for?

**Comment by Zerryth** [comment]
To be honest, haven't seen any tickets myself that raised an alarm of this being a huuuuge issue. I imagine most people who are adopting QnA Maker are using v3.0+. Only noticed it because I was writing tests

But it should be addressed "at some point". I don't know if we want to have a warning to update to v3.0+ (faster fix) or to actually be able to convert v2.0 (discussion needed on behavior of converting)

**Comment by Zerryth** [comment]
As per discussion with Chris, deprecating use of `v2.0` QnA Maker.
Will write fix to prompt user to use updated version
